libs/pages/services/src/lib/feature/page-settings-preview-environments-feature/page-settings-preview-environments-feature.tsx

```diff
@@ -80,27 +78,34 @@ export function PageSettingsPreviewEnvironmentsFeature() {
     }
   })
 
-  useEffect(() => {
+  const toggleAll = (value: boolean) => {
     //set all preview applications "true" when env preview is true
     if (loadingStatusEnvironmentDeploymentRules === 'loaded') {
-      applications?.forEach((application) => methods.setValue(application.id, watchEnvPreview))
+      applications?.forEach((application) => methods.setValue(application.id, value, { shouldDirty: true }))
     }
-  }, [loadingStatusEnvironmentDeploymentRules, watchEnvPreview, methods, applications])
+  }
 
   useEffect(() => {
     if (loadingStatusEnvironment === 'loaded') dispatch(fetchEnvironmentDeploymentRules(environmentId))
   }, [dispatch, loadingStatusEnvironment, environmentId])
 
   useEffect(() => {
-    if (loadingStatusEnvironmentDeploymentRules === 'loaded') {
+    // !loading is here to prevent the toggle to glitch the time we are submitting the two api endpoints
+    // previously the first endpoint updating the environment was making us come back to this useEffect
```

Comment:
nice that helped me understand! but maybe it would be good to remove this comment now?